addons/sourcemod/scripting/saxtonhale.sp

```diff
@@ -3003,6 +3007,22 @@ stock void TF2_Explode(int iAttacker = -1, float flPos[3], float flDamage, float
 		SDKHooks_TakeDamage(iBomb, 0, iAttacker, 9999.0);
 }
 
+stock void TF2_Shake(float flOrigin[3], float flAmplitude, float flRadius, float flDuration, float flFrequency)
+{
+	int iShake = CreateEntityByName("env_shake");
+	if (iShake != -1)
+	{
+		DispatchKeyValueVector(iShake, "origin", flOrigin);
+		DispatchKeyValueFloat(iShake, "amplitude", flAmplitude);
+		DispatchKeyValueFloat(iShake, "radius", flRadius);
+		DispatchKeyValueFloat(iShake, "duration", flDuration);
+		DispatchKeyValueFloat(iShake, "frequency", flFrequency);
+		
+		DispatchSpawn(iShake);
+		AcceptEntityInput(iShake, "StartShake");
```

Comment:
Need to kill entity once shake is done. Dunno if can be done in same frame to keep shake effect, otherwise timer may be needed with `flDuration`.